### PR TITLE
dbeaver/pro#2247 Close idle connections

### PR DESCRIPTION
**Pull request by serjiokov**
Dedicated attributes added for setup auto closable value(period) for connection configuration based on templates of connection types

### CODE REVIEW

**Comment by serjiokov** [comment]
image

**Comment by serjiokov** [comment]
image

File: plugins/org.jkiss.dbeaver.core/src/org/jkiss/dbeaver/core/CoreMessages.java [hunk 1]
```diff
@@ -53,6 +53,9 @@ public class CoreMessages extends NLS {
     public static String action_menu_transaction_auto_close_enabled;
     public static String action_menu_transaction_auto_close_enabled_tip;
     public static String action_menu_transaction_auto_close_ttl_tip;
```
Comment (serge-rider) [inline plugins/org.jkiss.dbeaver.core/src/org/jkiss/dbeaver/core/CoreMessages.java]: We already have this localization (dialog_connection_wizard_final_label_close_idle_connections)

**Review by serge-rider — CHANGES_REQUESTED** [review]

File: plugins/org.jkiss.dbeaver.core/src/org/jkiss/dbeaver/core/CoreMessages.java (hunk 1 above)
Comment (serjiokov) [reply plugins/org.jkiss.dbeaver.core/src/org/jkiss/dbeaver/core/CoreMessages.java]: These dedicated prefereces page localization
image

File: plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/connection/DBPConnectionType.java [hunk 2]
```diff
@@ -274,6 +292,22 @@ public void setCloseIdleTransactionPeriod(long closeIdleTransactionPeriod) {
         this.closeIdleTransactionPeriod = closeIdleTransactionPeriod;
     }
 
+    public boolean isAutoCloseConnections() {
```
Comment (serge-rider) [inline plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/connection/DBPConnectionType.java:295]: This option is never used anywhere

File: plugins/org.jkiss.dbeaver.core/src/org/jkiss/dbeaver/ui/dialogs/connection/ConnectionPageInitialization.java [hunk 3]
```diff
@@ -349,7 +350,11 @@ public void saveSettings(DBPDataSourceContainer dataSource) {
         bootstrap.setInitQueries(bootstrapQueries);
 
         confConfig.setKeepAliveInterval(keepAliveInterval.getSelection());
```
Comment (serge-rider) [inline plugins/org.jkiss.dbeaver.core/src/org/jkiss/dbeaver/ui/dialogs/connection/ConnectionPageInitialization.java:360]: You mustn't set connection settings from connection type.
DBPConnectionType must be used in runtime (as any other DBPConnectionType  properties)

**Review by serge-rider — CHANGES_REQUESTED** [review]

File: plugins/org.jkiss.dbeaver.core/src/org/jkiss/dbeaver/ui/dialogs/connection/ConnectionPageInitialization.java (hunk 3 above)
Comment (serjiokov) [reply plugins/org.jkiss.dbeaver.core/src/org/jkiss/dbeaver/ui/dialogs/connection/ConnectionPageInitialization.java:360]: Get it, m I right that require new logic for close idle connections in class DataSourceMonitorJob ?

**Comment by Matvey16** [comment]
Connection type properies are not being saved

https://github.com/dbeaver/dbeaver/assets/82543000/74a2180f-ec83-4032-9d03-baed0260e533

**Comment by Matvey16** [comment]
Verified

**Pull request merged**